File: Year-2023/LeetCode-April-Challenge/RemovingStarsFromaString.java
class Solution {
    public String removeStars(String s) {
        int i = 0, j = 0, n = s.length();
        char[] ch = s.toCharArray();
        while(j != n){
            if(ch[j] == '*') i--;
            else ch[i++] = ch[j];
            j++;
        }
        String res = new String(ch);
        return res.substring(0, i);
    }
}
public class RemovingStarsFromaString{
    public static void main(String[] args) {
        String  s = "leet**cod*e";
        System.out.println(new Solution().removeStars(s));
    }
}

File: Year-2023/LeetCode-April-Challenge/RemovingStarsFromaString.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    string removeStars(string s) {
        int i = 0, j = 0, n = s.size();
        while(j != n){
            if(s[j] == '*') i--;
            else s[i++] = s[j];
            j++;
        }
        return s.substr(0, i);
    }
};
int main(){
    string s = "leet**cod*e";
    Solution *S = new Solution();
    cout<<S->removeStars(s)<<endl;
    return 0;
}